test/extensions/filters/http/cdn_loop/parser_test.cc

```diff
@@ -0,0 +1,455 @@
+#include "extensions/filters/http/cdn_loop/parser.h"
+
+#include "test/test_common/status_utility.h"
+
+#include "absl/status/status.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace CdnLoop {
+namespace Parser {
+namespace {
+
+using ::Envoy::StatusHelpers::IsOkAndHolds;
+using ::Envoy::StatusHelpers::StatusIs;
+
+TEST(ParseOptionalWhitespaceTest, TestEmpty) {
+  const std::string value = "";
+  ParseContext input(value);
+  EXPECT_EQ(parseOptionalWhitespace(input), (ParseContext(value, 0)));
+}
+
+TEST(ParseOptionalWhitespaceTest, TestSpace) {
+  const std::string value = " ";
+  ParseContext input(value);
+  EXPECT_EQ(parseOptionalWhitespace(input), (ParseContext(value, 1)));
+}
+
+TEST(ParseOptionalWhitespaceTest, TestTab) {
+  const std::string value = "\t";
+  ParseContext input(value);
+  EXPECT_EQ(parseOptionalWhitespace(input), (ParseContext(value, 1)));
+}
+
+TEST(ParseOptionalWhitespaceTest, TestLots) {
+  const std::string value = "   \t \t ";
+  ParseContext input(value);
+  EXPECT_EQ(parseOptionalWhitespace(input), (ParseContext(value, 7)));
+}
+
+TEST(ParseOptionalWhitespaceTest, NoWhitespace) {
+  const std::string value = "c";
+  ParseContext input(value);
+  EXPECT_EQ(parseOptionalWhitespace(input), (ParseContext(value, 0)));
+}
+
+TEST(ParseOptionalWhitespaceTest, StopsAtNonWhitespace) {
+  const std::string value = "  c";
+  ParseContext input(value);
+  EXPECT_EQ(parseOptionalWhitespace(input), (ParseContext(value, 2)));
+}
+
+TEST(ParseQuotedPairTest, Simple) {
+  const std::string value = R"(\a)";
+  ParseContext input(value);
+  EXPECT_THAT(parseQuotedPair(input), IsOkAndHolds(ParseContext(value, 2)));
+}
+
+TEST(ParseQuotedPairTest, EndOfInput) {
+  const std::string value = R"(a)";
```

Comment:
What was this one testing? Looks like MissingBackslash

Reply:
It was supposed to be testing against empty string.  Fixed!